apps/etl/extraction/sources/base/handler.py

```diff
@@ -265,12 +265,12 @@ def _extraction_store_data(
         return extraction_object
 
     def _extraction_fetch_url(
-        self,
-        url: str,
-        headers: dict[str, typing.Any] | None = None,
+        self, url: str, headers: dict[str, typing.Any] | None = None, data: dict | None = None, method: str = "get"
     ) -> bool:
-        response = requests.get(url, headers=headers, timeout=30)
-
+        if method == "get":
+            response = requests.get(url, headers=headers, timeout=30)
+        elif method == "post":
+            response = requests.post(url, headers=headers, data=data, timeout=30)
```

Comment:
Let's add typing.assert_never(method) for type validation
```suggestion
        if method == "get":
            response = requests.get(url, headers=headers, timeout=30)
        elif method == "post":
            response = requests.post(url, headers=headers, data=data, timeout=30)
        else:
            typing.assert_never(method)
```